### ISSUE TITLE
Week 2 Tasks

### ISSUE THREAD

**Issue by sushmitabajgain** [sushmitabajgain/BLAM#2]
- [ ] **Core Model Implementations**
  - [ ] Implement `ContinuousAE` in `src/models.py`
  - [ ] Implement `VQVAE` following references.
  - [ ] Implement `BinaryAE` with basic binarization
  - [ ] Start Creating  prototyping.

- [ ] **Basic Training Pipeline**
  - [ ] Create training loop in 
  - [ ] Implement basic loss functions (MSE, reconstruction)
  - [ ] Test all models and train on MNIST
